x/leverage/keeper/borrows.go

```diff
@@ -12,152 +11,70 @@ import (
 func (k Keeper) GetBorrow(ctx sdk.Context, borrowerAddr sdk.AccAddress, denom string) sdk.Coin {
 	store := ctx.KVStore(k.storeKey)
 	owed := sdk.NewCoin(denom, sdk.ZeroInt())
-	key := types.CreateLoanKey(borrowerAddr, denom)
+	key := types.CreateAdjustedBorrowKey(borrowerAddr, denom)
 
+	adjustedAmount := sdk.ZeroDec()
 	if bz := store.Get(key); bz != nil {
-		err := owed.Amount.Unmarshal(bz)
+		err := adjustedAmount.Unmarshal(bz)
 		if err != nil {
 			panic(err)
 		}
 	}
 
+	// Apply interest scalar
+	owed.Amount = adjustedAmount.Mul(k.getInterestScalar(ctx, denom)).TruncateInt()
```

Comment:
Is truncating a concern here? Meaning, would rounding inefficiencies yield any potential issues?

Reply:
I'll look into it - especially in cases where a repayment or some other operation might leave tiny borrowed amount remainders instead of zero, since that could cause issues withdrawing collateral for example.

I suspect that for any real borrowed amount other than zero, the truncation shouldn't make any practical difference, but I can doublecheck.

Reply:
After checking into it:

- SetBorrow issues around zero won't be an issue, since it calls `setAdjustedBorrow(scalar * 0)` and thus doesn't leave a remainder

- For nonzero scenarios, all functions use `GetBorrow` and `SetBorrow`, all treat the truncated value as the "actual" amount, so they are not inconsistent with each other. Rounding errors never exceed one indivisible unit of token (e.g. `uatom` for Atom).

- The sum of all `AdjustedBorrow` is `TotalAdjustedBorrowed` for a denom - the `sdk.Dec` forms are always in sync (no error) but truncation could cause the sum of all `GetBorrow` to differ from `GetTotalBorrowed`. Each borrow position can contribute a maximum of 1 indivisible unit (e.g. `uatom`) to the sum of all `GetBorrow`. (Between those two approaches, `GetTotalBorrowed` is considered the correct one and has the minimum error).

- Derived quantity `BorrowUtilization` already defends against edge cases (e.g. 1000.1 / 1000) by returning a maximum of 1.0, and further quantities computed from utilization, like `LendAPY`, inherit that defense.

So at the moment, I don't see any truncation issues looming.

Reply:
perfect! Noted on preference of `GetTotalBorrowed` over summing over `GetBorrow ` 👍